CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/icon/IconDataNGTest.java

```diff
@@ -0,0 +1,540 @@
+        iconDataStaticMock.close();
+        imageIOStaticMock.close();
+    }
+
+    /**
+     * Test of getData method, of class IconData, when size not default and
+     * color not null.
+     */
+    @Test
+    public void testGetData() {
+        System.out.println("testGetData");
+
+        final IconData instance = spy(new IconDataImpl());
+
+        final int size = 1;
+        final Color color = new Color(1, 1, 1);
+
+        final byte[] expResult = new byte[0];
+        final byte[] result = instance.getData(size, color);
+
+        verify(instance, times(1)).createData(Mockito.eq(size), Mockito.eq(color));
+
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of getData method, of class IconData, when size is default and color
+     * not null.
+     */
+    @Test
+    public void testGetData_sizeIsDefault() {
+        System.out.println("testGetData_sizeIsDefault");
+
+        final IconData instance = spy(new IconDataImpl());
+
+        final int size = ConstellationIcon.DEFAULT_ICON_SIZE;
+        final Color color = new Color(1, 1, 1);
+
+        final byte[] expResult = new byte[0];
+        final byte[] result = instance.getData(size, color);
+
+        verify(instance, times(1)).createData(Mockito.eq(size), Mockito.eq(color));
+
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of getData method, of class IconData, when size not default and
+     * color is null.
+     */
+    @Test
+    public void testGetData_colorIsNull() {
+        System.out.println("testGetData_colorIsNull");
+
+        final IconData instance = spy(new IconDataImpl());
+
+        final int size = 1;
+        final Color color = null;
+
+        final byte[] expResult = new byte[0];
+        final byte[] result = instance.getData(size, color);
+
+        verify(instance, times(1)).createData(Mockito.eq(size), Mockito.eq(color));
+
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of getData method, of class IconData, when data is null.
+     */
+    @Test
+    public void testGetData_dataIsNull() {
+        System.out.println("testGetData_dataIsNull");
+
+        final IconData instance = spy(new IconDataImpl());
+
+        final int size = ConstellationIcon.DEFAULT_ICON_SIZE;
+        final Color color = null;
+
+        final byte[] expResult = new byte[0];
+        final byte[] result = instance.getData(size, color);
+
+        verify(instance, times(1)).createData(Mockito.eq(size), Mockito.eq(color));
+
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of getData method, of class IconData, when data not null.
+     */
+    @Test
+    public void testGetData_dataNotNull() {
+        System.out.println("testGetData_dataNotNull");
+
+        final IconData instance = spy(new IconDataImpl());
+
+        final int size = ConstellationIcon.DEFAULT_ICON_SIZE;
+        final Color color = null;
+
+        // Data not null.
+        instance.setData(new byte[1]);
```

Comment:
In this instance, instead of using `setData` you could call `getData` twice and verify that `createData` is only called the 1st time (in which case you could potentially merge with the test above). I don't think the exact size of data should make a difference here